Downgrade flex 2.6.4 to 2.6.3 to make it compile with GCC 7.3.0-16ubuntu3 (Ubuntu 18.04.1)

Otherwise it crashes with

    ./stage1flex   -o stage1scan.c ./scan.l
    Makefile:1701: recipe for target 'stage1scan.c' failed
    make[5]: *** [stage1scan.c] Segmentation fault (core dumped)

There are various patches which try to fix this crash but none
is working, so we downgrade the host tool and wait for an official
fix in the next release.

diff -Naur buildroot-2017.08.x_orig/package/flex/flex.hash buildroot-2017.08.x/package/flex/flex.hash
--- buildroot-2017.08.x_orig/package/flex/flex.hash	2018-10-15 12:57:41.219670156 +0200
+++ buildroot-2017.08.x/package/flex/flex.hash	2018-10-16 07:22:10.138167679 +0200
@@ -1,2 +1,3 @@
 # Locally computed:
 sha256 e87aae032bf07c26f85ac0ed3250998c37621d95f8bd748b31f15b33c45ee995  flex-2.6.4.tar.gz
+sha256 68b2742233e747c462f781462a2a1e299dc6207401dac8f0bbb316f48565c2aa  flex-2.6.3.tar.gz
diff -Naur buildroot-2017.08.x_orig/package/flex/flex.mk buildroot-2017.08.x/package/flex/flex.mk
--- buildroot-2017.08.x_orig/package/flex/flex.mk	2018-10-15 12:57:41.219670156 +0200
+++ buildroot-2017.08.x/package/flex/flex.mk	2018-10-16 07:45:01.378154397 +0200
@@ -4,8 +4,8 @@
 #
 ################################################################################
 
-FLEX_VERSION = 2.6.4
-FLEX_SITE = https://github.com/westes/flex/files/981163
+FLEX_VERSION = 2.6.3
+FLEX_SITE = https://github.com/westes/flex/releases/download/v$(FLEX_VERSION)
 FLEX_INSTALL_STAGING = YES
 FLEX_LICENSE = FLEX
 FLEX_LICENSE_FILES = COPYING
